### ISSUE TITLE
y

### ISSUE THREAD

**Issue by minkku** [minkku/java_project#15]
<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
<%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
<!DOCTYPE html>
<html>
  <head>
    <title>Vegefoods - Free Bootstrap 4 Template by Colorlib</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    
    <link href="https://fonts.googleapis.com/css?family=Poppins:200,300,400,500,600,700,800&display=swap" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Lora:400,400i,700,700i&display=swap" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Amatic+SC:400,700&display=swap" rel="stylesheet">

    <link rel="stylesheet" href="../resources/css/open-iconic-bootstrap.min.css">
    <link rel="stylesheet" href="../resources/css/animate.css">
    
    <link rel="stylesheet" href="../resources/css/owl.carousel.min.css">
    <link rel="stylesheet" href="../resources/css/owl.theme.default.min.css">
    <link rel="stylesheet" href="../resources/css/magnific-popup.css">

    <link rel="stylesheet" href="../resources/css/aos.css">

    <link rel="stylesheet" href="../resources/css/ionicons.min.css">

    <link rel="stylesheet" href="../resources/css/bootstrap-datepicker.css">
    <link rel="stylesheet" href="../resources/css/jquery.timepicker.css">

    
    <link rel="stylesheet" href="../resources/css/flaticon.css">
    <link rel="stylesheet" href="../resources/css/icomoon.css">
    <link rel="stylesheet" href="../resources/css/style.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
  </head>
	  <body class="goto-here">
		<div class="py-1 bg-primary">
    	<div class="container">
    		<div class="row no-gutters d-flex align-items-start align-items-center px-md-0">
	    		<div class="col-lg-12 d-block">
		    		<div class="row d-flex">
		    			<div class="col-md pr-4 d-flex topper align-items-center">
					    	<div class="icon mr-2 d-flex justify-content-center align-items-center"><span class="icon-phone2"></span></div>
						    <span class="text">+ 1235 2355 98</span>
					    </div>
					    <div class="col-md pr-4 d-flex topper align-items-center">
					    	<div class="icon mr-2 d-flex justify-content-center align-items-center"><span class="icon-paper-plane"></span></div>
						    <span class="text">youremail@email.com</span>
					    </div>
					    <div class="col-md-5 pr-4 d-flex topper align-items-center text-lg-right">
						    <span class="text">3-5 Business days delivery &amp; Free Returns</span>
					    </div>
				    </div>
			    </div>
		    </div>
		  </div>
    </div>

**Comment by minkku** [comment]
<%@ page language="java" contentType="text/html; charset=UTF-8"
	pageEncoding="UTF-8"%>
<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
<%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt"%>

 <nav class="navbar navbar-expand-lg navbar-dark ftco_navbar bg-dark ftco-navbar-light" id="ftco-navbar">
	    <div class="container">
	      <a class="navbar-brand" href="/sara">Vegefoods</a>
	      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#ftco-nav" aria-controls="ftco-nav" aria-expanded="false" aria-label="Toggle navigation">
	        <span class="oi oi-menu"></span> Menu
	      </button>

	      <div class="collapse navbar-collapse" id="ftco-nav">
	        <ul class="navbar-nav ml-auto">
	        	<c:if test="${empty signin}">
					<li class="nav-item active"><a href="/signin"
						class="nav-link"> Login</a></li>
				</c:if>
				<c:if test="${!empty signin}">
			 		<li class="nav-item dropdown">
              			<a class="nav-link dropdown-toggle" href="#" id="dropdown04" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">${signin.user_name}님
							안녕하세요.</a>
             			 <div class="dropdown-menu" aria-labelledby="dropdown04">
             			 	<a class="dropdown-item" href="/userInfo"> <i	class="fi fi-rr-user">내 정보</i></a>
              				<a class="dropdown-item" href="/board/mylist"><i class="fi fi-br-list">내 게시물</i></a>
              				<a class="dropdown-item" href="/orders/list"><i	class="fi fi-br-rectangle-list">주문 조회</i></a>
            			    <a class="dropdown-item"href="/carts"><i class="fi fi-rr-shopping-cart">장바구니</i></a>
              				<a class="dropdown-item"href="/logout" class="nav-link"><i class="fi fi-br-exit">로그아웃</i></a>
             			 </div>
             			 </li>
           			 </c:if>
			 
	         <li class="nav-item dropdown">
              <a class="nav-link dropdown-toggle" href="#" id="dropdown04" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Shop</a>
              <div class="dropdown-menu" aria-labelledby="dropdown04">
              	<a class="dropdown-item"ref="/book/allList">전체 보기</a>
              	<a class="dropdown-item" href="/book/edulist">education</a>
              	<a class="dropdown-item" href="/book/ficlist">fiction</a>
              	<a class="dropdown-item" href="/book/colist">comics</a>
                <a class="dropdown-item" href="checkout.html">Checkout</a>
              </div>
            </li>
	          
	           
	         <li class="nav-item dropdown">
              <a class="nav-link dropdown-toggle" href="#" id="dropdown04" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">board</a>
              <div class="dropdown-menu" aria-labelledby="dropdown04">
              <a class="dropdown-item" href="/board/notice">공지사항</a>
              <a class="dropdown-item" href="/board/list">게시판</a>
              </div>
              </li>
	          <li class="nav-item"><a href="/contact" class="nav-link">Contact</a></li>
	          <li class="nav-item"><a href="/about" class="nav-link">About</a></li>
	          <li class="nav-item cta cta-colored"><a href="cart.html" class="nav-link"><span class="icon-shopping_cart"></span>[0]</a></li>
	        </ul>
	      </div>
	    </div>
	  </nav>

**Comment by minkku** [comment]
@GetMapping("/modify")
public String getModify(@RequestParam("board_id") int board_id, Model model) {
    log.info("modify");
    
    // 필요한 경우 세션에서 users_id를 가져오도록 수정
    // HttpSession session 불필요
    // model.addAttribute("users_id", session.getAttribute("users_id"));

    log.info(board_id + "-----------board_id값");
    
    // 수정: get 메서드가 null을 반환할 경우에 대한 처리 추가
    BoardVO board = service.get(board_id);
    if (board == null) {
        // 적절한 오류 처리를 수행하거나, 에러 페이지로 리다이렉트할 수 있음
        // 여기서는 간단히 에러 메시지를 추가하여 리다이렉트
        model.addAttribute("error", "게시글을 찾을 수 없습니다.");
        return "redirect:/board/list";
    }
    
    model.addAttribute("board", board);
    return "/board/modify"; // 수정이 필요한 화면의 경로로 변경
}

@PostMapping("/modify")
public String modify(BoardVO board, RedirectAttributes rttr) {
    log.info("modify:" + board);
    
    // 수정: modify 메서드에서의 유효성 검사 추가
    if (board == null || board.getId() == 0) {
        // 적절한 오류 처리를 수행하거나, 에러 페이지로 리다이렉트할 수 있음
        // 여기서는 간단히 에러 메시지를 추가하여 리다이렉트
        rttr.addFlashAttribute("error", "수정할 게시글을 찾을 수 없습니다.");
        return "redirect:/board/list";
    }

    if (service.modify(board)) {
        rttr.addFlashAttribute("result", "success");
    }
    return "redirect:/board/list";
}
